spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkSchemaUtil.java

```diff
@@ -299,4 +302,17 @@ public static long estimateSize(StructType tableSchema, long totalRecords) {
     }
     return result;
   }
+
+  public static void validateMetadataColumnReferences(Schema tableSchema, Schema readSchema) {
```

Comment:
I initially did this for Spark only. It may make sense to move it to `DataTableScan` in `core`.